servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/RangeIntPublisher.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright © 2020 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.concurrent.api;
+
+import io.servicetalk.concurrent.PublisherSource;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static io.servicetalk.concurrent.internal.FlowControlUtils.addWithOverflowProtection;
+import static io.servicetalk.concurrent.internal.SubscriberUtils.isRequestNValid;
+import static io.servicetalk.concurrent.internal.SubscriberUtils.newExceptionForInvalidRequestN;
+import static java.lang.Math.min;
+import static java.util.Objects.requireNonNull;
+
+final class RangeIntPublisher extends AbstractSynchronousPublisher<Integer> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RangeIntPublisher.class);
+    private final int begin;
+    private final int end;
+    private final int stride;
+
+    RangeIntPublisher(int begin, int end) {
+        this(begin, end, 1);
+    }
+
+    RangeIntPublisher(int begin, int end, int stride) {
+        if (begin > end) {
+            throw new IllegalArgumentException("begin(" + begin + ") > end(" + end + ")");
+        }
+        if (stride <= 0) {
+            throw new IllegalArgumentException("stride: " + stride);
+        }
+        this.begin = begin;
+        this.end = end;
+        this.stride = stride;
+    }
+
+    @Override
+    void doSubscribe(final Subscriber<? super Integer> subscriber) {
+        subscriber.onSubscribe(new RangeIntSubscription(subscriber));
+    }
+
+    private final class RangeIntSubscription implements PublisherSource.Subscription {
+        private final PublisherSource.Subscriber<? super Integer> subscriber;
+        private long pendingN;
+        private int index = RangeIntPublisher.this.begin;
```

Comment:
nit: consider passing begin into the constructor so you can make the class static.

Reply:
the inner class also needs to access `end` and `stride` from the outer class. I will leave as is for now.